grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/HttpStreamDeframer.java

```diff
@@ -119,9 +119,16 @@ public void processHeaders(HttpHeaders headers, StreamDecoderOutput<DeframedMess
             try {
                 decompressor(ForwardingDecompressor.forGrpc(decompressor));
             } catch (Throwable t) {
-                final Metadata metadata = new Metadata();
-                transportStatusListener.transportReportStatus(exceptionHandler.apply(ctx, null, t, metadata),
-                                                              metadata);
+                Metadata metadata = Status.trailersFromThrowable(t);
+                if (metadata == null) {
+                    metadata = new Metadata();
+                }
+                Status grpcStatus0 = Status.fromThrowable(t);
+                if (grpcStatus0.getCause() != null) {
+                    grpcStatus0 = exceptionHandler.apply(ctx, grpcStatus0, t, metadata);
+                }
```

Comment:
I could see many duplicate codes with this.
How about moving this logic into `UnwrappingGrpcExceptionHandlerFunction` and reverting relevant changes?

`Status` will be nullable only for `UnwrappingGrpcExceptionHandlerFunction` but other handlers won't

Reply:
Then that violates the nullable convention. 
Let me create a util class and dedup the code there.  Thanks!

Reply:
Originally, `UnwrappingGrpcExceptionHandleFunction` was added for preprocessing before applying user-defined exception handlers so that we can directly invoke `GrpcExceptionHandlerFunction`.

If we use a utility class for preprocessing, `UnwrappingGrpcExceptionHandlerFunction` seems to be unnecessary.
Should we remove `UnwrappingGrpcExceptionHandlerFunction`?

Reply:
> Should we remove UnwrappingGrpcExceptionHandlerFunction and perform the logic in the util?

That's a good suggestion. 👍